mono_repo/lib/src/ci_shared.dart

```diff
@@ -14,6 +16,60 @@ String createdWith() => Zone.current[skipCreatedWithSentinel] == true
     ? ''
     : '# Created with package:mono_repo v$packageVersion\n';
 
+class CIJobEntry {
+  final CIJob job;
+  final List<String> commands;
+  final bool merge;
+
+  CIJobEntry(this.job, this.commands, this.merge);
+
+  String jobName(List<String> packages) {
+    final pkgLabel = packages.length == 1 ? 'PKG' : 'PKGS';
+
+    return 'SDK: ${job.sdk}; $pkgLabel: ${packages.join(', ')}; '
+        'TASKS: ${job.name}';
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      other is CIJobEntry &&
+      _equality.equals(_identityItems, other._identityItems);
+
+  @override
+  int get hashCode => _equality.hash(_identityItems);
+
+  List get _identityItems => [job.os, job.stageName, job.sdk, commands, merge];
```

Comment:
It seems like we should probably at least have a comment here explaining why we don't use all the fields from the CIJob (or its ==/hashCode)

Reply:
You made me do a bigger fix here, which is good.